volatility3/framework/plugins/linux/pslist.py

```diff
@@ -57,7 +78,37 @@ def _generator(self):
             if task.parent:
                 ppid = task.parent.pid
             name = utility.array_to_string(task.comm)
-            yield (0, (pid, ppid, name))
+            uid = task.real_cred.uid.val
+            gid = task.real_cred.gid.val
```

Comment:
Two things:

1) Why is 'real_cred' being used here instead of 'cred'? Volatility 2 uses 'cred', but I am now questioning which one is correct based on some reading:

- https://lore.kernel.org/lkml/20080806153907.14351.98173.stgit@warthog.procyon.org.uk/
- https://www.kernel.org/doc/html/latest/security/credentials.html

I reached out to a colleague who is a Linux kernel expert to see what he thinks, but I am curious if you had a reason for this also.

2)

Accessing .val directly will not work across all versions as the format/type of uid and gid changed quite a bit over the years. You can see how Volatility 2 handled all the variations here:

https://gitlab.research.volexity.com/Volcano/volatility_volc/-/blob/master/volatility/plugins/overlays/linux/linux.py#L1367

Regardless of if we use real_cred or cred, the handling of retrieving the value will need to match the flexibility of Volatility 2 in order to support all kernel versions needed.